Examples/Debugging/WhereIsThisPainting/components/PaintingInfo.js

```diff
@@ -0,0 +1,108 @@
+import React, {Component} from 'react';
+import {TouchableHighlight, View, Text, StyleSheet} from 'react-native';
+import API from '../lib/api';
+
+class PaintingInfo extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isPressed: false,
+      id: this.props.id,
+      info: {},
+    };
+  }
+
+  async componentDidMount() {
+    let info = await API.get(`/objects/${this.state.id}`, {
+      params: {
+        results: 1,
+        inc: 'primaryImage,title,artistDisplayName,objectEndDate,repository',
+      },
+    });
+    info = info.data;
+
+    const imageURL = `${info.primaryImage}`;
+    const title = `${info.title}`;
+    const artist = `${info.artistDisplayName}`;
+    const year = `${info.objectEndDate}`;
+    const location = `${info.repository}`;
+
+    this.setState({
+      ...this.state,
+      info: {
+        imageURL,
+        title,
+        artist,
+        year,
+        location,
+      },
+    });
+  }
+
+  _onPressIn = () => {
+    this.state.isPressed
+      ? this.setState({isPressed: false})
+      : this.setState({isPressed: true});
+  };
+
+  _onPressOut = () => {
+    this.state.isPressed
+      ? this.setState({isPressed: true})
+      : this.setState({isPressed: false});
+  };
```

Comment:
I think it can be handled by one function. Like `_onPressToggle = () => {}`

Reply:
It's reflected!